overlord/snapstate/handlers.go

```diff
@@ -118,11 +118,14 @@ func snapSetupAndState(t *state.Task) (*SnapSetup, *SnapState, error) {
 const defaultCoreSnapName = "core"
 const defaultBaseSnapsChannel = "stable"
 
-func changeInFlight(st *state.State, snapName string) (bool, error) {
+func changeInFlight(st *state.State, myChgID string, snapName string) (bool, error) {
 	for _, chg := range st.Changes() {
 		if chg.Status().Ready() {
 			continue
 		}
+		if myChgID == chg.ID() {
```

Comment:
I don't understand why we need this change? don't we end up adding the same thing again otherwise?

Reply:
This point makes me realize that there's a good chance we were talking across each other good part of the time in our call today.